Tank.java

```diff
@@ -1 +1,459 @@
+					bL = true;
+					break;
+
+				case KeyEvent.VK_W:
+					bU = true;
+					break;
+
+				case KeyEvent.VK_S:
+					bD = true;
+					break;
+			}}
+		if (player==2){
+			switch(key){
+				case KeyEvent.VK_RIGHT:
+					bR = true;
+					break;
+
+				case KeyEvent.VK_LEFT:
+					bL = true;
+					break;
+
+				case KeyEvent.VK_UP:
+					bU = true;
+					break;
+
+				case KeyEvent.VK_DOWN:
+					bD = true;
+					break;
+			}
+		}
+		decideDirection();
+	}
+
+	void decideDirection() {
+		if (!bL && !bU && bR && !bD)
+			direction = Direction.R;
+
+		else if (bL && !bU && !bR && !bD)
+			direction = Direction.L;
+
+		else if (!bL && bU && !bR && !bD)
+			direction = Direction.U;
+
+		else if (!bL && !bU && !bR && bD)
+			direction = Direction.D;
+
+		else if (!bL && !bU && !bR && !bD)
+			direction = Direction.STOP;
+	}
+
+	public void keyReleased(KeyEvent e) {
+		int key = e.getKeyCode();
+		if (player==1){
+			switch (key) {
+
+				case KeyEvent.VK_F:
+					fire();
+					break;
+
+				case KeyEvent.VK_D:
+					bR = false;
+					break;
+
+				case KeyEvent.VK_A:
+					bL = false;
+					break;
+
+				case KeyEvent.VK_W:
+					bU = false;
+					break;
+
+				case KeyEvent.VK_S:
+					bD = false;
+					break;
+
+
+			}}
+		if (player==2){
+			switch (key) {
+
+				case KeyEvent.VK_SLASH:
+					fire();
+					break;
+
+				case KeyEvent.VK_RIGHT:
+					bR = false;
+					break;
+
+				case KeyEvent.VK_LEFT:
+					bL = false;
+					break;
+
+				case KeyEvent.VK_UP:
+					bU = false;
+					break;
+
+				case KeyEvent.VK_DOWN:
+					bD = false;
+					break;
+
+
+			}}
+		decideDirection();
+	}
+
+	public Bullets fire() {
+		if (!live)
+			return null;
+		int X = this.x + Tank.WIDTH / 2 - Bullets.width / 2;
+		int Y = this.y + Tank.LENGHT / 2 - Bullets.length / 2;
+		Bullets m = new Bullets(X, Y + 2, good, myDirection, this.tc);
+		tc.bullets.add(m);
+		return m;
+	}
+
+
+	public Rectangle getRect() {
+		return new Rectangle(x, y, WIDTH, LENGHT);
+	}
+
+	public boolean isLive() {
+		return live;
+	}
+
+	public void setLive(boolean live) {
+		this.live = live;
+	}
+
+	public boolean isGood() {
+		return good;
+	}
+
+	public boolean collideWithWall(CommonWall w) {
+		if (this.live && this.getRect().intersects(w.getRect())) {
+			this.changToOldDir();
+			return true;
+		}
+		return false;
+	}
+
+	public boolean collideWithWall(MetalWall w) {
+		if (this.live && this.getRect().intersects(w.getRect())) {
+			this.changToOldDir();
+			return true;
+		}
+		return false;
+	}
+
+	public boolean collideRiver(River r) {
+		if (this.live && this.getRect().intersects(r.getRect())) {
+			this.changToOldDir();
+			return true;
+		}
+		return false;
+	}
+
+	public boolean collideHome(Home h) {
+		if (this.live && this.getRect().intersects(h.getRect())) {
+			this.changToOldDir();
+			return true;
+		}
+		return false;
+	}
+
+	public boolean collideWithTanks(java.util.List<Tank> tanks) {
+		for (int i = 0; i < tanks.size(); i++) {
+			Tank t = tanks.get(i);
```

Comment:
collideWithTanks methode, naudoji traditional for-loop. Gali naudoti enhanced for-loop instead.
Pvz.:
for (Tank t : tanks) {
    if (this != t && this.live && t.isLive() && this.getRect().intersects(t.getRect())) {
        this.changToOldDir();
        t.changToOldDir();
        return true;
    }
}